core/src/main/java/com/klaytn/caver/tx/type/AbstractTxType.java

```diff
@@ -84,14 +83,56 @@ public AbstractTxType(BigInteger nonce, BigInteger gasPrice, BigInteger gasLimit
         this.from = from;
         this.to = to;
         this.value = value;
+        this.senderSignatureDataSet = new HashSet<>();
     }
 
+  @Deprecated
     protected void setSenderSignatureData(KlaySignatureData signatureData) {
-        this.senderSignatureData = signatureData;
+        addSenderSignatureData(signatureData);
     }
 
+    @Deprecated
     public KlaySignatureData getSenderSignatureData() {
-        return senderSignatureData;
+        Iterator<KlaySignatureData> senderSignatureIterator = senderSignatureDataSet.iterator();
+        if (senderSignatureIterator.hasNext()) {
+            return senderSignatureIterator.next();
+        } else {
+            return null;
+        }
+    }
+
+    protected void addSenderSignatureData(KlaySignatureData signatureData) {
+        senderSignatureDataSet.add(signatureData);
+    }
+
+    public void addSenderSignatureData(Set<KlaySignatureData> senderSignatureDataSet) {
+        this.senderSignatureDataSet.addAll(senderSignatureDataSet);
+    }
+
+    protected void addSenderSignatureData(List<RlpType> signatureRlpTypeList) {
```

Comment:
I do not understand why this parameter collection is `List` and the other method parameter collection is `Set`

Reply:
In decoding the encoded raw transaction, the signature can be obtained as a `List` of `RlpType`. Why should a parameter be a `Set`? `Set` is just a type that manages `senderSignature` in `AbstractTxType`.

Reply:
You're right. I misunderstood!